### PR TITLE
New version of ldap provisioner

### PR DESCRIPTION
**Pull request by lionick**
You must also execute the following queries:
ALTER TABLE cm_co_ldap_provisioner_targets ADD COLUMN attr_opts boolean;
ALTER TABLE cm_co_ldap_provisioner_targets ADD COLUMN hierarchy_format boolean;
ALTER TABLE cm_co_ldap_provisioner_targets  ADD COLUMN oc_voperson boolean;
ALTER TABLE cm_co_ldap_provisioner_targets  DROP COLUMN opt_lang;
ALTER TABLE cm_co_ldap_provisioner_targets  DROP COLUMN opt_role;

### CODE REVIEW

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]
Please add changes to CHANGELOG under `## [Unreleased]`

**Comment by lionick** [comment]
> Please add changes to CHANGELOG under `## [Unreleased]`

Done.

File: app/Plugin/LdapProvisioner/Model/CoLdapProvisionerDn.php [hunk 1]
```diff
@@ -87,8 +87,25 @@ public function assignGroupDn($coProvisioningTargetData, $coGroupData) {
       // Throw an exception... this should be defined
       throw new RuntimeException(_txt('er.ldapprovisioner.dn.config'));
     }
-    
-    $dn = "cn=" . $coGroupData['CoGroup']['name']
+    if($coGroupData['CoGroup']['cou_id'] != NULL && $coProvisioningTargetData['CoLdapProvisionerTarget']['hierarchy_format']) {
```
Comment (ioigoume) [inline app/Plugin/LdapProvisioner/Model/CoLdapProvisionerDn.php]: perhaps use this: !is_null($coGroupData['CoGroup']['cou_id'])

File: app/Plugin/LdapProvisioner/View/CoLdapProvisionerTargets/fields.inc [hunk 2]
```diff
@@ -293,16 +300,18 @@
...
               $ocopts['disabled'] = true;
             }
-
+            if(!empty($co_ldap_provisioner_targets[0]['CoLdapProvisionerTarget']['oc_' . strtolower($oc)])){
+              $ocopts['checked'] = true;
+            }
             print $this->Form->input('oc_' . strtolower($oc), $ocopts) . "<br />";
 
-            print '<div id="oc_' . $oc . '_div">';
+            print '<div id="oc_' . $oc . '_div" class="nested">';
```
Comment (ioigoume) [inline app/Plugin/LdapProvisioner/View/CoLdapProvisionerTargets/fields.inc:314]: class nested is missing from co-base.css? Have you tested how ui will behave if you import that rule?

**Review by ioigoume — CHANGES_REQUESTED** [review]

**Pull request merged**